File: hospital/SergioDeIscar-hospital-java/SergioDeIscar-hospital-java/src/main/java/org/example/models/Paciente.java
package org.example.models;

import org.example.enums.TipoPaciente;

import java.time.LocalDate;

public final class Paciente {
    private final String dni;
    private final String nombre;
    private final LocalDate fechaNacimiento;
    private final TipoPaciente tipo;


    private LocalDate fechaIngreso = null;
    private LocalDate fechaAlta = null;

    public Paciente(String dni, String nombre, LocalDate fechaNacimiento, TipoPaciente tipo) {
        this.dni = dni;
        this.nombre = nombre;
        this.fechaNacimiento = fechaNacimiento;
        this.tipo = tipo;
    }
    public Paciente(String dni, String nombre, TipoPaciente tipo) {
        this.dni = dni;
        this.nombre = nombre;
        this.fechaNacimiento = LocalDate.now();
        this.tipo = tipo;
    }
    public String getDni() {
        return dni;
    }
    public String getNombre() {
        return nombre;
    }
    public LocalDate getFechaNacimiento() {
        return fechaNacimiento;
    }
    public TipoPaciente getTipo() {
        return tipo;
    }
    public LocalDate getFechaIngreso() {
        return fechaIngreso;
    }
    public LocalDate getFechaAlta() {
        return fechaAlta;
    }
    public void setFechaIngreso(LocalDate fechaIngreso) {
        this.fechaIngreso = fechaIngreso;
    }
    public void setFechaAlta(LocalDate fechaAlta) {
        this.fechaAlta = fechaAlta;
    }

    @Override
    public int hashCode() {
        return dni.hashCode() + nombre.hashCode() + tipo.hashCode() + fechaNacimiento.hashCode();
    }

    @Override
    public boolean equals(Object obj) {
        return obj instanceof Paciente &&
                ((Paciente) obj).getDni().equals(dni) &&
                ((Paciente) obj).getNombre().equals(nombre) &&
                ((Paciente) obj).getTipo().equals(tipo) &&
                ((Paciente) obj).getFechaNacimiento().equals(fechaNacimiento);
    }
}

File: hospital/SergioDeIscar-hospital-kotlin/SergioDeIscar-hospital-kotlin/src/main/kotlin/models/Paciente.kt
package models

import enums.TipoPaciente
import java.time.LocalDate

class Paciente(val dni: String, val nombre: String, val fechaNacimiento: LocalDate = LocalDate.now(), val tipo: TipoPaciente) {
    var fechaIgreso: LocalDate = LocalDate.now()
        private set
    var fechaAlta: LocalDate = LocalDate.now()
        private set

    fun setFechaIngreso(date: LocalDate){
        fechaIgreso = date
    }
    fun setFechaAlta(date: LocalDate){
        fechaAlta = date
    }

    override fun hashCode(): Int {
        return dni.hashCode() + fechaNacimiento.hashCode() + fechaIgreso.hashCode() + fechaAlta.hashCode()
    }

    override fun equals(other: Any?): Boolean {
        return other is Paciente && other.dni == dni && other.fechaIgreso == fechaIgreso && other.fechaAlta == fechaAlta
    }
}